engine/src/main/java/org/terasology/engine/ComponentSystemManager.java

```diff
@@ -84,7 +84,14 @@ public void loadSystems(ModuleEnvironment environment, NetworkMode netMode) {
             Name moduleId = environment.getModuleProviding(type);
             RegisterSystem registerInfo = type.getAnnotation(RegisterSystem.class);
             if (registerInfo.value().isValidFor(netMode, isHeadless)) {
-                systemsByModule.put(moduleId, type);
+                if(registerInfo.requiresOptional().length != 0){
+                    for(String s : registerInfo.requiresOptional()){
+                        if(environment.get(new Name(s))!=null)systemsByModule.put(moduleId, type);
+                    }
```

Comment:
woudln't it make sense to have the system register only if all dependencies are present?

(Also I am not sure if this is a good concept in general, but since some people want it  i won't object)